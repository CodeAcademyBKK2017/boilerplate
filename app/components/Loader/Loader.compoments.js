import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Loader.compoments.style';
import {ActivityIndicator, Modal, View
} from 'react-native';

export default class Loader extends Component {
  render () {
    return (
      <Modal transparent={true} visible={this.props.show} style={[styles.container, styles.horizontal]}>
        <View  style={styles.containerView}>
          <ActivityIndicator animating={this.props.show} size='large' color='#0000ff' />
        </View>
      </Modal>
    );
  }
}

Loader.propTypes = {
  show: PropTypes.bool
};
  