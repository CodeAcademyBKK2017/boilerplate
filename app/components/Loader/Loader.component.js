import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Loader.style';
import {
  ActivityIndicator,
  Modal,
  View
} from 'react-native';

class Loader extends Component {
  render () {
    return (
      <Modal
        visible={this.props.visible}
        transparent={this.props.transparent}
        animationType={'fade'}
      >
        <View style={[styles.container, styles.horizontal]}>
          <ActivityIndicator size='large' color='#cc0000' />
        </View>
      </Modal>
    );
  }
}

Loader.propTypes = {
  visible: PropTypes.bool,
  transparent: PropTypes.bool
};
  
Loader.defaultProps = {
  visible: 'true',
  transparent: 'false'
};

export default Loader;