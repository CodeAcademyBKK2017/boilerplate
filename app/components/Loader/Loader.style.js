import {
  StyleSheet
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#00000033'
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10
  }
});

export default styles;
  