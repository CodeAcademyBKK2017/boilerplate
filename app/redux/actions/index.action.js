import {createAction} from 'redux-actions';

export const ADD_NOTE = 'ADD_NOTE';
export const DELETE_NOTE = 'DELETE_NOTE';
export const POPULATE_NOTE = 'POPULATE_NOTE';
export const SHOW_LOADER = 'SHOW_LOADER';
export const HIDE_LOADER = 'HIDE_LOADER';

export const addnote = createAction(ADD_NOTE);
export const deletenote = createAction(DELETE_NOTE);
export const populatenote = createAction(POPULATE_NOTE);
export const showloader = createAction(SHOW_LOADER);
export const hideloader = createAction(SHOW_LOADER);