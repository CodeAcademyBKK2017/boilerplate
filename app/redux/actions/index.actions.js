import {createAction} from 'redux-actions';

// notes

export const ADD_NOTE = 'ADD_NOTE';
export const DELETE_NOTE = 'DELETE_NOTE';
export const POPULATE_NOTES = 'POPULATE_NOTES';

export const FETCH_NOTES = 'FETCH_NOTES';
export const SAVE_NOTE = 'SAVE_NOTE';

export const addNote = createAction(ADD_NOTE);
export const deleteNote = createAction(DELETE_NOTE);
export const populateNotes = createAction(POPULATE_NOTES);

export const fetchNotes = createAction(FETCH_NOTES);
export const saveNote = createAction(SAVE_NOTE);

// loader

export const SHOW_LOADER = 'SHOW_LOADER';
export const HIDE_LOADER = 'HIDE_LOADER';

export const showLoader = createAction(SHOW_LOADER);
export const hideLoader = createAction(HIDE_LOADER);
