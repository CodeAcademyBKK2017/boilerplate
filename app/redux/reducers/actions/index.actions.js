import {createAction} from 'redux-actions';

export const ADD_NOTE = 'ADD_NOTE';
export const DELE_NOTE = 'DELE_NOTE';
export const LOAD_SERVER = 'LOAD_SERVER';

export const addNote = createAction(ADD_NOTE);
export const deleNote = createAction(DELE_NOTE);
export const loadServer = createAction(LOAD_SERVER);