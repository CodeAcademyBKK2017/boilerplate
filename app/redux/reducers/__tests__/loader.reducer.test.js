import Loader from '../../../components/Loader/Loader.component';
import LoaderReducer from '../loader.reducer';
import React from 'react';
import renderer from 'react-test-renderer';

import * as actions from '../../actions/index.actions';

describe('Loader', () => {
  it('Snapshot', () => {
    const snapshot = renderer.create(<Loader visible={true} transparent={true} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Reducer Show', () => {
    const showLoaderAction = {
      type: actions.SHOW_LOADER
    };
    expect(LoaderReducer({}, showLoaderAction)).toEqual(true);
  });

  it('Reducer Hide', () => {
    const hideLoaderAction = {
      type: actions.HIDE_LOADER
    };
    expect(LoaderReducer({}, hideLoaderAction)).toEqual(false);
  });

  it('Reducer Default', () => {
    const defaultLoaderAction = {
      type: {}
    };
    expect(LoaderReducer(false, defaultLoaderAction)).toEqual(false);
  });

  it('Reducer Undefined', () => {
    const undefinedLoaderAction = {
      type: {}
    };
    expect(LoaderReducer(undefined, undefinedLoaderAction)).toEqual(false);
  });
});
