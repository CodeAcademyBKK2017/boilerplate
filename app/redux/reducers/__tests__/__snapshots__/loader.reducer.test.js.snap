// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#00000033",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#cc0000"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</Modal>
`;
